R/pkg/R/DataFrame.R

```diff
@@ -1795,17 +1795,15 @@ setMethod("fillna",
               if (length(colNames) == 0 || !all(colNames != "")) {
                 stop("value should be an a named list with each name being a column name.")
               }
-
-              # Convert to the named list to an environment to be passed to JVM
-              valueMap <- new.env()
-              for (col in colNames) {
-                # Check each item in the named list is of valid type
-                v <- value[[col]]
+              # Check each item in the named list is of valid type
+              lapply(value, function(v) {
```

Comment:
nit: this is going through it twice..  perhaps add a FUN param to `convertNamedListToEnv` to allow for a custom validation for each value?